src/appshell/view/preferences/appearancepreferencesmodel.cpp

```diff
@@ -131,6 +183,17 @@ ThemeList AppearancePreferencesModel::allThemes() const
     return uiConfiguration()->themes();
 }
 
+bool AppearancePreferencesModel::isCurrentThemeHighContrast() const
+{
+    return uiConfiguration()->currentTheme().codeKey == HIGH_CONTRAST_BLACK_THEME_CODE
```

Comment:
Better to simplify this method like this (the same for isCurrentThemeGeneral):

static const QStringList contrastThemes {
HIGH_CONTRAST_BLACK_THEME_CODE,
HIGH_CONTRAST_WHITE_THEME_CODE
};

return contrastThemes.contais(currentThemeCode());

Reply:
UPD: seems like isCurrentThemeHighContrast and isCurrentThemeGeneral were copy-pasted in several places. Better to use UiConfiguration::isCurrentThemeHighContrast() (and UiConfiguration::isCurrentThemeGeneral()) and apply my suggested fixes there

Reply:
> Better to simplify this method like this

This is how the code was turning out to be. 

```c++
bool UiConfiguration::isCurrentThemeHighContrast() const
{
    static const QStringList highContrastThemes {
        QString::fromStdString(HIGH_CONTRAST_BLACK_THEME_CODE),
        QString::fromStdString(HIGH_CONTRAST_WHITE_THEME_CODE)
    };

    return highContrastThemes.contains(QString::fromStdString(currentThemeCodeKey()));
}
```

To try to reduce the overheads and conversions, i tried to implement it in a way that used `std::string` itself, and came up with this:
```c++
static const std::vector<std::string> highContrastThemes {
        HIGH_CONTRAST_BLACK_THEME_CODE,
        HIGH_CONTRAST_WHITE_THEME_CODE
    };
    
    if(std::find(highContrastThemes.begin(), highContrastThemes.end(), currentThemeCodeKey()) != highContrastThemes.end())
    {
        return true;
    } else {
        return false;
    }
```

Should i go with this instead?

Reply:
In this case, I more like the first implementation for this method :) We can just slightly improve it:

ThemeCode currentThemeCode = uiConfiguration()->currentTheme().codeKey;
return currentThemeCode == HIGH_CONTRAST_BLACK_THEME_CODE || currentThemeCode == HIGH_CONTRAST_WHITE_THEME_CODE;

Reply:
> In this case, I more like the first implementation for this method :)
You mean the original implementation, I believe?

Since this function has now found a new home in `uiconfiguration.cpp`, this is what it looks like now, after all suggested changes:
```c++
bool UiConfiguration::isCurrentThemeGeneral() const
{
    ThemeCode currentThemeCode = currentThemeCodeKey();

    return currentThemeCode == DARK_THEME_CODE || currentThemeCode == LIGHT_THEME_CODE;
}
```

and

```
bool UiConfiguration::isCurrentThemeHighContrast() const
{
    ThemeCode currentThemeCode = currentThemeCodeKey();

    return currentThemeCode == HIGH_CONTRAST_BLACK_THEME_CODE || currentThemeCode == HIGH_CONTRAST_WHITE_THEME_CODE;
}
```